src/atoms/Link/Link.native.js

```diff
@@ -0,0 +1,71 @@
+import React, { useContext, useState, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import Text from '../Text';
+import styled, { ThemeContext } from 'styled-components/native';
+
+const StyledTouchable = styled.TouchableOpacity`
+  align-self: flex-start;
+`;
+
+const resolveSizeToTextProps = (linkSize) => {
+  //TODO: Add lineHeight prop for all cases
+  switch (linkSize) {
+    case 's':
+      return { size: 'xs', weight: 'regular' };
+    case 'm':
+      return { size: 'm', weight: 'regular' };
+    case 'l':
+      return { size: 'l', weight: 'regular' };
+    default:
+      return { size: 'xs', weight: 'regular' };
+  }
+};
+
+const Link = ({ children, visited, onClick, size }) => {
+  const theme = useContext(ThemeContext);
+  const [active, setActive] = useState(false);
+
+  const onPressIn = useCallback(() => {
+    setActive(true);
+  }, [setActive]);
+  const onPressOut = useCallback(() => {
+    setActive(false);
+  }, [setActive]);
+
+  const inactiveTextColor = visited ? theme.colors.secondary[900] : theme.colors.primary[800];
+  const activeTextColor = visited ? theme.colors.secondary[600] : theme.colors.primary[600];
+
+  return (
+    <StyledTouchable
+      activeOpacity={1}
+      onPress={onClick}
+      onPressIn={onPressIn}
+      onPressOut={onPressOut}
+      testID="ds-link-button"
+    >
+      <Text
+        underline={active}
+        color={active ? activeTextColor : inactiveTextColor}
+        {...resolveSizeToTextProps(size)}
+      >
+        {children}
+      </Text>
+    </StyledTouchable>
+  );
+};
+
+Link.propTypes = {
+  visited: PropTypes.bool,
+  children: PropTypes.node.isRequired,
+  onClick: PropTypes.func,
+  size: PropTypes.oneOf(['s', 'm', 'l']),
```

Comment:
we need to now discuss whether we should call all these `s/m/l` variants as `size` prop or `type` prop. Since it's already different in Text and now here Link 😆